agent/src/beerocks/slave/son_slave_thread.cpp

```diff
@@ -4475,37 +4475,50 @@ bool slave_thread::handle_channel_selection_request(Socket *sd, ieee1905_1::Cmdu
                          });
```

Comment:
You are again correct. The current code assumes that `preference` is always `0`, meaning always non-operable. The correct solution is to take the actual preference into account but then we will need to add much more code which is not related to this PR and should be handled in #725 - this I prefer not to do as part of a bug fix PR.

Reply:
Just to clarify, this comment was on the upper for-loop and not the one you have changed.

Reply:
Just a quick note because I've seen it before, this is not correct:
> Adding everything as a reference is useless and it is translated to passing everything possible as a reference to the function"

Only the variables you use in the lambda body will be captured by reference so you don't need to explicitly list them (see [cppreference](https://en.cppreference.com/w/cpp/language/lambda)). 
Of course you can still list them if you prefer but if it's for "performance reasons", then you don't have to. The 2 are equivalent.